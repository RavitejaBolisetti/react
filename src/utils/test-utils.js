/* eslint-disable import/no-extraneous-dependencies */
import { ReactElement } from "react";
import { render, RenderOptions } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import { Provider } from "react-redux";
import { configureStore } from "@store/configureStore";
const store = configureStore({});
const AllTheProviders = ({ children }) => {
  return (
    <MemoryRouter>
      <Provider store={store}>{children}</Provider>;
    </MemoryRouter>
  );
};

const customRender = (ui, options) => render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from "@testing-library/react";

export default customRender;
