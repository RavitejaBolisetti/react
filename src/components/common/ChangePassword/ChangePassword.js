import React, { useState, useEffect } from 'react';
import { connect } from 'react-redux';
import { Button, Col, Form, Input, Modal, Row } from 'antd';
import { bindActionCreators } from 'redux';
import { changePasswordActions } from 'store/actions/data/changePassword';
import { validateFieldsPassword, validateRequiredInputField } from 'utils/validation';
import { handleErrorModal, handleSuccessModal } from 'utils/responseModal';

const mapStateToProps = (state) => {
    const {
        auth: { token, isLoggedIn, userId },

        data: {
            ChangePassword: { isLoading, isLoaded: isDataLoaded = false },
        },
    } = state;

    return {
        isDataLoaded,
        token,
        isLoggedIn,
        userId,
        isLoading,
    };
};

const mapDispatchToProps = (dispatch) => ({
    dispatch,
    ...bindActionCreators(
        {
            saveData: changePasswordActions.saveData,
            listShowLoading: changePasswordActions.listShowLoading,
        },
        dispatch
    ),
});

const ChangePasswordBase = ({ isOpen = false, onOk = () => {}, onCancel = () => {}, title = '', discreption = '', saveData, isDataLoaded, listShowLoading, userId }) => {
    const [form] = Form.useForm();

    const [confirmDirty, setConfirmDirty] = useState(false);

    const onFinish = (errorInfo) => {
              // form.validateFields().then((values) => {});
    };
    const onFinishFailed = (values) => {
        if(values.errorFields.length==0){
       const data = { ...values };
       const onSuccess = (res) => {
           form.resetFields();
           handleSuccessModal({ title: 'SUCCESS', message: res?.responseMessage });
       };

       const onError = (message) => {
           handleErrorModal(message);
       };

       const requestData = {
           data: [data],
           setIsLoading: listShowLoading,
           userId,
           onSuccess,
           onError
           
       };

       saveData(requestData); 
    }       
    };

    const validateToNextPassword = (rule, value, callback) => {
        if (value && confirmDirty) {
            form.validateFields(['confirmPassword'], { force: true });
        }
        callback();
    };

    const compareToFirstPassword = (rule, value, callback) => {
        if (value && value !== form.getFieldValue('newPassword')) {
            callback("New Password and Confirm Password doesn't match!");
        } else {
            callback();
        }
    };

    const handleConfirmBlur = (e) => {
        const value = e.target.value;
        setConfirmDirty(confirmDirty || !!value);
    };
    return (
        <>
            <Modal open={isOpen} title={title} okText="Submit" footer={false} okType="primary"  onOk={onFinishFailed} onCancel={onCancel}>
                {discreption ? (
                    <Row gutter={20}>
                        <Col xs={24} sm={24} md={24} lg={24} xl={24}>
                            <p>{discreption}</p>
                        </Col>
                    </Row>
                ) : (
                    ''
                )}
                <Form form={form} name="change_password" layout="vertical" autoComplete="false" onFinish={onFinish} onFinishFailed={onFinishFailed}>
                    <Row gutter={20}>
                        <Col xs={24} sm={24} md={24} lg={24} xl={24}>
                            <Form.Item label="Old Password" name="oldPassword" rules={[validateRequiredInputField('Old Password')]}>
                                <Input.Password type="text" allowClear placeholder="Enter Old Password" visibilityToggle={true} />
                            </Form.Item>
                        </Col>
                    </Row>
                    <Row gutter={20}>
                        <Col xs={24} sm={24} md={24} lg={24} xl={24}>
                            <Form.Item
                                label="New Password"
                                name="newPassword"
                                rules={[
                                    validateRequiredInputField('New Password'),
                                    validateFieldsPassword('New Password'),
                                    {
                                        validator: validateToNextPassword,
                                    },
                                ]}
                            >
                                <Input.Password type="text" allowClear placeholder="Enter New Password" visibilityToggle={true} />
                            </Form.Item>
                        </Col>
                    </Row>
                    <Row gutter={20}>
                        <Col xs={24} sm={24} md={24} lg={24} xl={24}>
                            <Form.Item
                                label="Confirm Password"
                                name="confirmPassword"
                                rules={[
                                    validateRequiredInputField('Confirm Password'),
                                    validateFieldsPassword('Confirm Password'),
                                    {
                                        validator: compareToFirstPassword,
                                    },
                                ]}
                            >
                                <Input.Password type="text" allowClear placeholder="Enter Confirm Password" onBlur={handleConfirmBlur} visibilityToggle={true} />
                            </Form.Item>
                        </Col>
                    </Row>
                    <Row gutter={20}>
                        <Col xs={12} sm={12} md={12} lg={12} xl={12}>
                            <Button style={{ marginTop: '20px' }}>Cancel</Button>
                        </Col>
                        <Col xs={12} sm={12} md={12} lg={12} xl={12} style={{ textAlign: 'right' }}>
                            <Button style={{ marginTop: '20px' }} type="primary" htmlType="submit">
                                Submit
                            </Button>
                        </Col>
                    </Row>
                </Form>
            </Modal>
        </>
    );
};

export const ChangePassword = connect(mapStateToProps, mapDispatchToProps)(ChangePasswordBase);
