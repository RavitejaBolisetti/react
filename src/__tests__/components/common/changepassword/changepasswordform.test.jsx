/*
 *   Copyright (c) 2023 Mahindra & Mahindra Ltd. 
 *   All rights reserved.
 *   Redistribution and use of any source or binary or in any form, without written approval and permission is prohibited. Please read the Terms of Use, Disclaimer & Privacy Policy on https://www.mahindra.com/
 */
import { screen, fireEvent } from "@testing-library/react";
import '@testing-library/jest-dom/extend-expect';
import { act } from 'react-dom/test-utils'
import { ChangePasswordForm } from "@components/common/ChangePassword/ChangePasswordForm";
import customRender from "@utils/test-utils";



// wrap the root component in a Provider
describe("ChangePasswordForm Components", () => {
    it("should render ChangePasswordForm components", ()=> {
      const changePasswordForm =  customRender(<ChangePasswordForm />);
      expect(changePasswordForm).toMatchSnapshot();
    });
    it("should change old password field in the screen", ()=>{
      customRender(<ChangePasswordForm />);
      expect(screen.getByText(/Old password/i)).toBeInTheDocument();
    })
    it("should change password field event", async () => {
          customRender(<ChangePasswordForm />);
          await act(async () => {
            const changePasswordButton = screen.getByRole('button', {
                name: /Change Password/i
              });
            fireEvent.click(changePasswordButton);      
          });
      });
});