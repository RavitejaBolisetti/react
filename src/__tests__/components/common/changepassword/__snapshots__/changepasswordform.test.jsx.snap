// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChangePasswordForm Components should render ChangePasswordForm components 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <form
        autocomplete="off"
        class="ant-form ant-form-vertical css-dev-only-do-not-override-w8mnev changePasswordForm"
        id="change_password"
      >
        <div
          class="ant-row css-dev-only-do-not-override-w8mnev"
          style="margin-left: -10px; margin-right: -10px;"
        >
          <div
            class="ant-col textfieldWithPrefix ant-col-xs-24 ant-col-sm-24 ant-col-md-24 ant-col-lg-24 ant-col-xl-24 css-dev-only-do-not-override-w8mnev"
            style="padding-left: 10px; padding-right: 10px;"
          >
            <div
              class="ant-form-item textfieldWithPrefix__input css-dev-only-do-not-override-w8mnev"
            >
              <div
                class="ant-row ant-form-item-row css-dev-only-do-not-override-w8mnev"
              >
                <div
                  class="ant-col ant-form-item-control css-dev-only-do-not-override-w8mnev"
                >
                  <div
                    class="ant-form-item-control-input"
                  >
                    <div
                      class="ant-form-item-control-input-content"
                    >
                      <span
                        class="ant-input-affix-wrapper css-dev-only-do-not-override-w8mnev"
                      >
                        <span
                          class="ant-input-prefix"
                        >
                          <svg
                            fill="none"
                            height="18"
                            stroke="currentColor"
                            stroke-linecap="round"
                            stroke-linejoin="round"
                            stroke-width="2"
                            viewBox="0 0 24 24"
                            width="18"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <rect
                              height="11"
                              rx="2"
                              ry="2"
                              width="18"
                              x="3"
                              y="11"
                            />
                            <path
                              d="M7 11V7a5 5 0 0 1 10 0v4"
                            />
                          </svg>
                        </span>
                        <input
                          aria-required="true"
                          class="ant-input css-dev-only-do-not-override-w8mnev"
                          data-testid="oldPasswordInput"
                          id="change_password_oldPassword"
                          type="password"
                          value=""
                        />
                        <span
                          class="ant-input-suffix"
                        >
                          <span>
                            <svg
                              fill="currentColor"
                              height="18"
                              stroke="currentColor"
                              stroke-width="0"
                              viewBox="0 0 1024 1024"
                              width="18"
                              xmlns="http://www.w3.org/2000/svg"
                            >
                              <path
                                d="M942.2 486.2Q889.47 375.11 816.7 305l-50.88 50.88C807.31 395.53 843.45 447.4 874.7 512 791.5 684.2 673.4 766 512 766q-72.67 0-133.87-22.38L323 798.75Q408 838 512 838q288.3 0 430.2-300.3a60.29 60.29 0 0 0 0-51.5zm-63.57-320.64L836 122.88a8 8 0 0 0-11.32 0L715.31 232.2Q624.86 186 512 186q-288.3 0-430.2 300.3a60.3 60.3 0 0 0 0 51.5q56.69 119.4 136.5 191.41L112.48 835a8 8 0 0 0 0 11.31L155.17 889a8 8 0 0 0 11.31 0l712.15-712.12a8 8 0 0 0 0-11.32zM149.3 512C232.6 339.8 350.7 258 512 258c54.54 0 104.13 9.36 149.12 28.39l-70.3 70.3a176 176 0 0 0-238.13 238.13l-83.42 83.42C223.1 637.49 183.3 582.28 149.3 512zm246.7 0a112.11 112.11 0 0 1 146.2-106.69L401.31 546.2A112 112 0 0 1 396 512z"
                              />
                              <path
                                d="M508 624c-3.46 0-6.87-.16-10.25-.47l-52.82 52.82a176.09 176.09 0 0 0 227.42-227.42l-52.82 52.82c.31 3.38.47 6.79.47 10.25a111.94 111.94 0 0 1-112 112z"
                              />
                            </svg>
                          </span>
                        </span>
                      </span>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <label
              class="textfieldWithPrefix__label"
            >
              Old password
            </label>
          </div>
        </div>
        <div
          class="ant-row css-dev-only-do-not-override-w8mnev"
          style="margin-left: -10px; margin-right: -10px;"
        >
          <div
            class="ant-col textfieldWithPrefix ant-col-xs-24 ant-col-sm-24 ant-col-md-24 ant-col-lg-24 ant-col-xl-24 css-dev-only-do-not-override-w8mnev"
            style="padding-left: 10px; padding-right: 10px;"
          >
            <div
              class="ant-form-item textfieldWithPrefix__input css-dev-only-do-not-override-w8mnev"
            >
              <div
                class="ant-row ant-form-item-row css-dev-only-do-not-override-w8mnev"
              >
                <div
                  class="ant-col ant-form-item-control css-dev-only-do-not-override-w8mnev"
                >
                  <div
                    class="ant-form-item-control-input"
                  >
                    <div
                      class="ant-form-item-control-input-content"
                    >
                      <span
                        class="ant-input-affix-wrapper css-dev-only-do-not-override-w8mnev"
                      >
                        <span
                          class="ant-input-prefix"
                        >
                          <svg
                            fill="none"
                            height="18"
                            stroke="currentColor"
                            stroke-linecap="round"
                            stroke-linejoin="round"
                            stroke-width="2"
                            viewBox="0 0 24 24"
                            width="18"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <rect
                              height="11"
                              rx="2"
                              ry="2"
                              width="18"
                              x="3"
                              y="11"
                            />
                            <path
                              d="M7 11V7a5 5 0 0 1 10 0v4"
                            />
                          </svg>
                        </span>
                        <input
                          aria-required="true"
                          class="ant-input css-dev-only-do-not-override-w8mnev"
                          data-testid="newPasswordInput"
                          id="change_password_newPassword"
                          type="password"
                          value=""
                        />
                        <span
                          class="ant-input-suffix"
                        >
                          <span>
                            <svg
                              fill="currentColor"
                              height="18"
                              stroke="currentColor"
                              stroke-width="0"
                              viewBox="0 0 1024 1024"
                              width="18"
                              xmlns="http://www.w3.org/2000/svg"
                            >
                              <path
                                d="M942.2 486.2Q889.47 375.11 816.7 305l-50.88 50.88C807.31 395.53 843.45 447.4 874.7 512 791.5 684.2 673.4 766 512 766q-72.67 0-133.87-22.38L323 798.75Q408 838 512 838q288.3 0 430.2-300.3a60.29 60.29 0 0 0 0-51.5zm-63.57-320.64L836 122.88a8 8 0 0 0-11.32 0L715.31 232.2Q624.86 186 512 186q-288.3 0-430.2 300.3a60.3 60.3 0 0 0 0 51.5q56.69 119.4 136.5 191.41L112.48 835a8 8 0 0 0 0 11.31L155.17 889a8 8 0 0 0 11.31 0l712.15-712.12a8 8 0 0 0 0-11.32zM149.3 512C232.6 339.8 350.7 258 512 258c54.54 0 104.13 9.36 149.12 28.39l-70.3 70.3a176 176 0 0 0-238.13 238.13l-83.42 83.42C223.1 637.49 183.3 582.28 149.3 512zm246.7 0a112.11 112.11 0 0 1 146.2-106.69L401.31 546.2A112 112 0 0 1 396 512z"
                              />
                              <path
                                d="M508 624c-3.46 0-6.87-.16-10.25-.47l-52.82 52.82a176.09 176.09 0 0 0 227.42-227.42l-52.82 52.82c.31 3.38.47 6.79.47 10.25a111.94 111.94 0 0 1-112 112z"
                              />
                            </svg>
                          </span>
                        </span>
                      </span>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <label
              class="textfieldWithPrefix__label"
            >
              New password
            </label>
          </div>
        </div>
        <div
          class="ant-row css-dev-only-do-not-override-w8mnev"
          style="margin-left: -10px; margin-right: -10px;"
        >
          <div
            class="ant-col textfieldWithPrefix ant-col-xs-24 ant-col-sm-24 ant-col-md-24 ant-col-lg-24 ant-col-xl-24 css-dev-only-do-not-override-w8mnev"
            style="padding-left: 10px; padding-right: 10px;"
          >
            <div
              class="ant-form-item textfieldWithPrefix__input css-dev-only-do-not-override-w8mnev"
            >
              <div
                class="ant-row ant-form-item-row css-dev-only-do-not-override-w8mnev"
              >
                <div
                  class="ant-col ant-form-item-control css-dev-only-do-not-override-w8mnev"
                >
                  <div
                    class="ant-form-item-control-input"
                  >
                    <div
                      class="ant-form-item-control-input-content"
                    >
                      <span
                        class="ant-input-affix-wrapper css-dev-only-do-not-override-w8mnev"
                      >
                        <span
                          class="ant-input-prefix"
                        >
                          <svg
                            fill="none"
                            height="18"
                            stroke="currentColor"
                            stroke-linecap="round"
                            stroke-linejoin="round"
                            stroke-width="2"
                            viewBox="0 0 24 24"
                            width="18"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <rect
                              height="11"
                              rx="2"
                              ry="2"
                              width="18"
                              x="3"
                              y="11"
                            />
                            <path
                              d="M7 11V7a5 5 0 0 1 10 0v4"
                            />
                          </svg>
                        </span>
                        <input
                          aria-required="true"
                          class="ant-input css-dev-only-do-not-override-w8mnev"
                          data-testid="confirmNewPasswordInput"
                          id="change_password_confirmNewPassword"
                          type="password"
                          value=""
                        />
                        <span
                          class="ant-input-suffix"
                        >
                          <span>
                            <svg
                              fill="currentColor"
                              height="18"
                              stroke="currentColor"
                              stroke-width="0"
                              viewBox="0 0 1024 1024"
                              width="18"
                              xmlns="http://www.w3.org/2000/svg"
                            >
                              <path
                                d="M942.2 486.2Q889.47 375.11 816.7 305l-50.88 50.88C807.31 395.53 843.45 447.4 874.7 512 791.5 684.2 673.4 766 512 766q-72.67 0-133.87-22.38L323 798.75Q408 838 512 838q288.3 0 430.2-300.3a60.29 60.29 0 0 0 0-51.5zm-63.57-320.64L836 122.88a8 8 0 0 0-11.32 0L715.31 232.2Q624.86 186 512 186q-288.3 0-430.2 300.3a60.3 60.3 0 0 0 0 51.5q56.69 119.4 136.5 191.41L112.48 835a8 8 0 0 0 0 11.31L155.17 889a8 8 0 0 0 11.31 0l712.15-712.12a8 8 0 0 0 0-11.32zM149.3 512C232.6 339.8 350.7 258 512 258c54.54 0 104.13 9.36 149.12 28.39l-70.3 70.3a176 176 0 0 0-238.13 238.13l-83.42 83.42C223.1 637.49 183.3 582.28 149.3 512zm246.7 0a112.11 112.11 0 0 1 146.2-106.69L401.31 546.2A112 112 0 0 1 396 512z"
                              />
                              <path
                                d="M508 624c-3.46 0-6.87-.16-10.25-.47l-52.82 52.82a176.09 176.09 0 0 0 227.42-227.42l-52.82 52.82c.31 3.38.47 6.79.47 10.25a111.94 111.94 0 0 1-112 112z"
                              />
                            </svg>
                          </span>
                        </span>
                      </span>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <label
              class="textfieldWithPrefix__label"
            >
              Confirm password
            </label>
          </div>
        </div>
        <div
          class="ant-row css-dev-only-do-not-override-w8mnev"
          style="margin-left: -10px; margin-right: -10px;"
        >
          <div
            class="ant-col textCenter ant-col-xs-24 ant-col-sm-24 ant-col-md-24 ant-col-lg-24 ant-col-xl-24 css-dev-only-do-not-override-w8mnev"
            style="padding-left: 10px; padding-right: 10px;"
          >
            <button
              class="ant-btn css-dev-only-do-not-override-w8mnev ant-btn-primary"
              data-testid="changePassword"
              style="margin-top: 6px;"
              type="submit"
            >
              <span>
                Change Password
              </span>
            </button>
          </div>
        </div>
      </form>
      ;
    </div>
  </body>,
  "container": <div>
    <form
      autocomplete="off"
      class="ant-form ant-form-vertical css-dev-only-do-not-override-w8mnev changePasswordForm"
      id="change_password"
    >
      <div
        class="ant-row css-dev-only-do-not-override-w8mnev"
        style="margin-left: -10px; margin-right: -10px;"
      >
        <div
          class="ant-col textfieldWithPrefix ant-col-xs-24 ant-col-sm-24 ant-col-md-24 ant-col-lg-24 ant-col-xl-24 css-dev-only-do-not-override-w8mnev"
          style="padding-left: 10px; padding-right: 10px;"
        >
          <div
            class="ant-form-item textfieldWithPrefix__input css-dev-only-do-not-override-w8mnev"
          >
            <div
              class="ant-row ant-form-item-row css-dev-only-do-not-override-w8mnev"
            >
              <div
                class="ant-col ant-form-item-control css-dev-only-do-not-override-w8mnev"
              >
                <div
                  class="ant-form-item-control-input"
                >
                  <div
                    class="ant-form-item-control-input-content"
                  >
                    <span
                      class="ant-input-affix-wrapper css-dev-only-do-not-override-w8mnev"
                    >
                      <span
                        class="ant-input-prefix"
                      >
                        <svg
                          fill="none"
                          height="18"
                          stroke="currentColor"
                          stroke-linecap="round"
                          stroke-linejoin="round"
                          stroke-width="2"
                          viewBox="0 0 24 24"
                          width="18"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <rect
                            height="11"
                            rx="2"
                            ry="2"
                            width="18"
                            x="3"
                            y="11"
                          />
                          <path
                            d="M7 11V7a5 5 0 0 1 10 0v4"
                          />
                        </svg>
                      </span>
                      <input
                        aria-required="true"
                        class="ant-input css-dev-only-do-not-override-w8mnev"
                        data-testid="oldPasswordInput"
                        id="change_password_oldPassword"
                        type="password"
                        value=""
                      />
                      <span
                        class="ant-input-suffix"
                      >
                        <span>
                          <svg
                            fill="currentColor"
                            height="18"
                            stroke="currentColor"
                            stroke-width="0"
                            viewBox="0 0 1024 1024"
                            width="18"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <path
                              d="M942.2 486.2Q889.47 375.11 816.7 305l-50.88 50.88C807.31 395.53 843.45 447.4 874.7 512 791.5 684.2 673.4 766 512 766q-72.67 0-133.87-22.38L323 798.75Q408 838 512 838q288.3 0 430.2-300.3a60.29 60.29 0 0 0 0-51.5zm-63.57-320.64L836 122.88a8 8 0 0 0-11.32 0L715.31 232.2Q624.86 186 512 186q-288.3 0-430.2 300.3a60.3 60.3 0 0 0 0 51.5q56.69 119.4 136.5 191.41L112.48 835a8 8 0 0 0 0 11.31L155.17 889a8 8 0 0 0 11.31 0l712.15-712.12a8 8 0 0 0 0-11.32zM149.3 512C232.6 339.8 350.7 258 512 258c54.54 0 104.13 9.36 149.12 28.39l-70.3 70.3a176 176 0 0 0-238.13 238.13l-83.42 83.42C223.1 637.49 183.3 582.28 149.3 512zm246.7 0a112.11 112.11 0 0 1 146.2-106.69L401.31 546.2A112 112 0 0 1 396 512z"
                            />
                            <path
                              d="M508 624c-3.46 0-6.87-.16-10.25-.47l-52.82 52.82a176.09 176.09 0 0 0 227.42-227.42l-52.82 52.82c.31 3.38.47 6.79.47 10.25a111.94 111.94 0 0 1-112 112z"
                            />
                          </svg>
                        </span>
                      </span>
                    </span>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <label
            class="textfieldWithPrefix__label"
          >
            Old password
          </label>
        </div>
      </div>
      <div
        class="ant-row css-dev-only-do-not-override-w8mnev"
        style="margin-left: -10px; margin-right: -10px;"
      >
        <div
          class="ant-col textfieldWithPrefix ant-col-xs-24 ant-col-sm-24 ant-col-md-24 ant-col-lg-24 ant-col-xl-24 css-dev-only-do-not-override-w8mnev"
          style="padding-left: 10px; padding-right: 10px;"
        >
          <div
            class="ant-form-item textfieldWithPrefix__input css-dev-only-do-not-override-w8mnev"
          >
            <div
              class="ant-row ant-form-item-row css-dev-only-do-not-override-w8mnev"
            >
              <div
                class="ant-col ant-form-item-control css-dev-only-do-not-override-w8mnev"
              >
                <div
                  class="ant-form-item-control-input"
                >
                  <div
                    class="ant-form-item-control-input-content"
                  >
                    <span
                      class="ant-input-affix-wrapper css-dev-only-do-not-override-w8mnev"
                    >
                      <span
                        class="ant-input-prefix"
                      >
                        <svg
                          fill="none"
                          height="18"
                          stroke="currentColor"
                          stroke-linecap="round"
                          stroke-linejoin="round"
                          stroke-width="2"
                          viewBox="0 0 24 24"
                          width="18"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <rect
                            height="11"
                            rx="2"
                            ry="2"
                            width="18"
                            x="3"
                            y="11"
                          />
                          <path
                            d="M7 11V7a5 5 0 0 1 10 0v4"
                          />
                        </svg>
                      </span>
                      <input
                        aria-required="true"
                        class="ant-input css-dev-only-do-not-override-w8mnev"
                        data-testid="newPasswordInput"
                        id="change_password_newPassword"
                        type="password"
                        value=""
                      />
                      <span
                        class="ant-input-suffix"
                      >
                        <span>
                          <svg
                            fill="currentColor"
                            height="18"
                            stroke="currentColor"
                            stroke-width="0"
                            viewBox="0 0 1024 1024"
                            width="18"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <path
                              d="M942.2 486.2Q889.47 375.11 816.7 305l-50.88 50.88C807.31 395.53 843.45 447.4 874.7 512 791.5 684.2 673.4 766 512 766q-72.67 0-133.87-22.38L323 798.75Q408 838 512 838q288.3 0 430.2-300.3a60.29 60.29 0 0 0 0-51.5zm-63.57-320.64L836 122.88a8 8 0 0 0-11.32 0L715.31 232.2Q624.86 186 512 186q-288.3 0-430.2 300.3a60.3 60.3 0 0 0 0 51.5q56.69 119.4 136.5 191.41L112.48 835a8 8 0 0 0 0 11.31L155.17 889a8 8 0 0 0 11.31 0l712.15-712.12a8 8 0 0 0 0-11.32zM149.3 512C232.6 339.8 350.7 258 512 258c54.54 0 104.13 9.36 149.12 28.39l-70.3 70.3a176 176 0 0 0-238.13 238.13l-83.42 83.42C223.1 637.49 183.3 582.28 149.3 512zm246.7 0a112.11 112.11 0 0 1 146.2-106.69L401.31 546.2A112 112 0 0 1 396 512z"
                            />
                            <path
                              d="M508 624c-3.46 0-6.87-.16-10.25-.47l-52.82 52.82a176.09 176.09 0 0 0 227.42-227.42l-52.82 52.82c.31 3.38.47 6.79.47 10.25a111.94 111.94 0 0 1-112 112z"
                            />
                          </svg>
                        </span>
                      </span>
                    </span>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <label
            class="textfieldWithPrefix__label"
          >
            New password
          </label>
        </div>
      </div>
      <div
        class="ant-row css-dev-only-do-not-override-w8mnev"
        style="margin-left: -10px; margin-right: -10px;"
      >
        <div
          class="ant-col textfieldWithPrefix ant-col-xs-24 ant-col-sm-24 ant-col-md-24 ant-col-lg-24 ant-col-xl-24 css-dev-only-do-not-override-w8mnev"
          style="padding-left: 10px; padding-right: 10px;"
        >
          <div
            class="ant-form-item textfieldWithPrefix__input css-dev-only-do-not-override-w8mnev"
          >
            <div
              class="ant-row ant-form-item-row css-dev-only-do-not-override-w8mnev"
            >
              <div
                class="ant-col ant-form-item-control css-dev-only-do-not-override-w8mnev"
              >
                <div
                  class="ant-form-item-control-input"
                >
                  <div
                    class="ant-form-item-control-input-content"
                  >
                    <span
                      class="ant-input-affix-wrapper css-dev-only-do-not-override-w8mnev"
                    >
                      <span
                        class="ant-input-prefix"
                      >
                        <svg
                          fill="none"
                          height="18"
                          stroke="currentColor"
                          stroke-linecap="round"
                          stroke-linejoin="round"
                          stroke-width="2"
                          viewBox="0 0 24 24"
                          width="18"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <rect
                            height="11"
                            rx="2"
                            ry="2"
                            width="18"
                            x="3"
                            y="11"
                          />
                          <path
                            d="M7 11V7a5 5 0 0 1 10 0v4"
                          />
                        </svg>
                      </span>
                      <input
                        aria-required="true"
                        class="ant-input css-dev-only-do-not-override-w8mnev"
                        data-testid="confirmNewPasswordInput"
                        id="change_password_confirmNewPassword"
                        type="password"
                        value=""
                      />
                      <span
                        class="ant-input-suffix"
                      >
                        <span>
                          <svg
                            fill="currentColor"
                            height="18"
                            stroke="currentColor"
                            stroke-width="0"
                            viewBox="0 0 1024 1024"
                            width="18"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <path
                              d="M942.2 486.2Q889.47 375.11 816.7 305l-50.88 50.88C807.31 395.53 843.45 447.4 874.7 512 791.5 684.2 673.4 766 512 766q-72.67 0-133.87-22.38L323 798.75Q408 838 512 838q288.3 0 430.2-300.3a60.29 60.29 0 0 0 0-51.5zm-63.57-320.64L836 122.88a8 8 0 0 0-11.32 0L715.31 232.2Q624.86 186 512 186q-288.3 0-430.2 300.3a60.3 60.3 0 0 0 0 51.5q56.69 119.4 136.5 191.41L112.48 835a8 8 0 0 0 0 11.31L155.17 889a8 8 0 0 0 11.31 0l712.15-712.12a8 8 0 0 0 0-11.32zM149.3 512C232.6 339.8 350.7 258 512 258c54.54 0 104.13 9.36 149.12 28.39l-70.3 70.3a176 176 0 0 0-238.13 238.13l-83.42 83.42C223.1 637.49 183.3 582.28 149.3 512zm246.7 0a112.11 112.11 0 0 1 146.2-106.69L401.31 546.2A112 112 0 0 1 396 512z"
                            />
                            <path
                              d="M508 624c-3.46 0-6.87-.16-10.25-.47l-52.82 52.82a176.09 176.09 0 0 0 227.42-227.42l-52.82 52.82c.31 3.38.47 6.79.47 10.25a111.94 111.94 0 0 1-112 112z"
                            />
                          </svg>
                        </span>
                      </span>
                    </span>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <label
            class="textfieldWithPrefix__label"
          >
            Confirm password
          </label>
        </div>
      </div>
      <div
        class="ant-row css-dev-only-do-not-override-w8mnev"
        style="margin-left: -10px; margin-right: -10px;"
      >
        <div
          class="ant-col textCenter ant-col-xs-24 ant-col-sm-24 ant-col-md-24 ant-col-lg-24 ant-col-xl-24 css-dev-only-do-not-override-w8mnev"
          style="padding-left: 10px; padding-right: 10px;"
        >
          <button
            class="ant-btn css-dev-only-do-not-override-w8mnev ant-btn-primary"
            data-testid="changePassword"
            style="margin-top: 6px;"
            type="submit"
          >
            <span>
              Change Password
            </span>
          </button>
        </div>
      </div>
    </form>
    ;
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
